src/power.jl

```diff
@@ -58,37 +58,95 @@ end
 
 ^(a::Taylor1{TaylorN{T}}, r::Rational) where {T<:NumberNotSeries} = a^(r.num/r.den)
 
+# in-place form of power_by_squaring
+# this method assumes `y`, `x` and `aux1` are of same order
+function power_by_squaring!(y::TaylorN{T}, x::TaylorN{T}, aux1::TaylorN{T}, p::Integer) where {T<:NumberNotSeries}
+    t = trailing_zeros(p) + 1
+    p >>= t
+    # aux1 = x
+    for k in eachindex(aux1)
+        identity!(aux1, x, k)
+    end
+    while (t -= 1) > 0
+        # aux1 = square(aux1)
+        for k in reverse(eachindex(aux1))
+            sqr!(aux1, k)
+        end
+    end
+    # y = aux1
+    for k in eachindex(y)
+        identity!(y, aux1, k)
+    end
+    while p > 0
+        t = trailing_zeros(p) + 1
+        p >>= t
+        while (t -= 1) ≥ 0
+            # aux1 = square(aux1)
+            for k in reverse(eachindex(aux1))
+                sqr!(aux1, k)
+            end
+        end
+        # y = y * aux1
+        mul!(y, aux1)
+    end
+    return nothing
+end
 
 
 # power_by_squaring; slightly modified from base/intfuncs.jl
 # Licensed under MIT "Expat"
 for T in (:Taylor1, :HomogeneousPolynomial, :TaylorN)
-    @eval function power_by_squaring(x::$T, p::Integer)
-        if p == 1
-            return copy(x)
-        elseif p == 0
-            return one(x)
-        elseif p == 2
-            return square(x)
-        end
-        t = trailing_zeros(p) + 1
-        p >>= t
-        while (t -= 1) > 0
-            x = square(x)
-        end
-        y = x
-        while p > 0
+    @eval power_by_squaring(x::$T, p::Integer) = power_by_squaring(x, Val(p))
+    @eval power_by_squaring(x::$T, ::Val{0}) = one(x)
+    @eval power_by_squaring(x::$T, ::Val{1}) = copy(x)
+    @eval power_by_squaring(x::$T, ::Val{2}) = square(x)
+    @eval function power_by_squaring(x::$T, ::Val{P}) where P
+        p = P # copy static parameter `P` into local variable `p`
+        if $T == TaylorN
+            y = zero(x)
+            aux1 = zero(x)
+            power_by_squaring!(y, x, aux1, p)
+        else
             t = trailing_zeros(p) + 1
             p >>= t
-            while (t -= 1) ≥ 0
+            while (t -= 1) > 0
                 x = square(x)
             end
-            y *= x
+            y = x
+            while p > 0
+                t = trailing_zeros(p) + 1
+                p >>= t
+                while (t -= 1) ≥ 0
+                    x = square(x)
+                end
+                y *= x
+            end
         end
         return y
     end
 end
 
+power_by_squaring(x::TaylorN{Taylor1{T}}, ::Val{0}) where {T<:NumberNotSeries} = one(x)
+power_by_squaring(x::TaylorN{Taylor1{T}}, ::Val{1}) where {T<:NumberNotSeries} = copy(x)
+power_by_squaring(x::TaylorN{Taylor1{T}}, ::Val{2}) where {T<:NumberNotSeries} = square(x)
+function power_by_squaring(x::TaylorN{Taylor1{T}}, ::Val{P}) where {P, T<:NumberNotSeries}
```

Comment:
What about `Taylor1{TaylorN{T}}` method? Or is it not necessary?

Reply:
You are right, so please ignore my comment. Once said this, i think there will be no changes in `@taylorize` that involve `power_by_squaring!`. I thus mark this comment as resolved.